packages/web/pages/smart-accounts/index.tsx

```diff
@@ -0,0 +1,424 @@
+import { fromBase64, toBase64 } from "@cosmjs/encoding";
+import { PrivKeySecp256k1 } from "@keplr-wallet/crypto";
+import { isNil } from "@osmosis-labs/utils";
+import { useMutation } from "@tanstack/react-query";
+import { observer } from "mobx-react-lite";
+import type { NextPage } from "next";
+import { FunctionComponent } from "react";
+
+import { Button } from "~/components/buttons";
+import { OsmoverseCard } from "~/components/cards/osmoverse-card";
+import { CheckBox } from "~/components/control";
+import Spinner from "~/components/spinner";
+import { EventName } from "~/config";
+import { useAmplitudeAnalytics } from "~/hooks";
+import { Authenticator } from "~/server/queries/osmosis/authenticators";
+import { useStore } from "~/stores";
+import { api, RouterOutputs } from "~/utils/trpc";
+
+const SmartAccounts: NextPage = observer(function () {
+  const { chainStore, accountStore } = useStore();
+  useAmplitudeAnalytics({
+    onLoadEvent: [EventName.Pools.pageViewed],
+  });
+  const { chainId } = chainStore.osmosis;
+  const account = accountStore.getWallet(chainId);
+
+  const osmosisAddress = account?.address ?? "";
+
+  const { data: authenticators, refetch: refetchAuthenticators } =
+    api.edge.authenticators.getAuthenticators.useQuery(
+      {
+        userOsmoAddress: osmosisAddress,
+      },
+      {
+        enabled: osmosisAddress.length > 0,
+      }
+    );
+
+  const { data: cosmosAccount } =
+    api.edge.authenticators.getCosmosAccount.useQuery(
+      {
+        address: osmosisAddress,
+      },
+      {
+        enabled: osmosisAddress.length > 0,
+      }
+    );
+
+  const removeAuthenticator = useMutation(async ({ id }: { id: string }) => {
```

Comment:
As a follow up from Slack discussion, would be good to extract this mutation into a separate file that can be reused for sending txs. Then we can reliable update all relevant queries when this mutation is triggered.